### PR TITLE
Fixed inference on Metal GPU backend + updated docs for run_* programs

### PR DESCRIPTION
**Pull request by svilupp**
- Added CHANGELOG.md file to keep track of changes (we can call it NEWS / HISTORY / ...)
- Updated docstrings for `run_llama`, `run_chat`, and `run_server` to be more informative.
- Changed default `run_*` parameters to be more sensible for first-time users, including the `run_server` port number (`port=10897`) to be unique and not clash with other services.
- Updated run context with the necessary files for fast inference on Metal GPUs (eg, Apple Macbooks M-series)

Default parameter changes:
- ngl=99 -> because it's 15x slower without it, so I'd rather push to GPU too much than too little
- num-ctx=2048 -> default is 512. Most models will have >=2048. We would artificially knee-cap models and users would find it hard to debug the reason (still risky, but less so)
- port=10897 -> to avoid clashes

Added README reference how to start the server.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+- Changed default `run_*` parameters to be more sensible for first-time users, including the `run_server` port number (`port=519`) to be unique and not clash with other services.
+- Updated run context with the necessary files for fast inference on Metal GPUs (eg, Apple Macbooks M-series)
+
+## [0.2.0]
+
+### Added
+- Added `run_server` functionality that starts a simple HTTP server (interact either in your browser or use other LLM packages). It provides an OpenAI-compatible chat completion endpoint.
+
+### Fixed
+- Updated llama.cpp JLL bindings to the latest version.
```
Comment (marcom) [inline CHANGELOG.md]: Probably best to state exactly which version

File: main.log
Comment (marcom) [inline main.log:1]: Is this file needed at all?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+## [Unreleased]
+
+### Added
+
+### Fixed
+- Updated docstrings for `run_llama`, `run_chat`, and `run_server` to be more informative.
+- Changed default `run_*` parameters to be more sensible for first-time users, including the `run_server` port number (`port=519`) to be unique and not clash with other services.
```
Comment (marcom) [inline CHANGELOG.md]: Port 519, while cute reasoning for the number :), is below 1024 and reserved (utime service). Normally on unix unprivileged users cannot listen on ports below 1024. So I would suggest to use a port above 1024.

File: main.log
Comment (svilupp) [reply main.log:1]: Removed! oversight

File: CHANGELOG.md (hunk 1 above)
Comment (svilupp) [reply CHANGELOG.md]: Good point!

**Comment by svilupp** [comment]
As part of this, I've noticed that `run_chat` is infinite :D CTRL+c control sequence gets eaten by REPL, so we can't kill it... Will open an issue

**Review by marcom — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (svilupp) [reply CHANGELOG.md]: hmm, good point! I looked through the wiki tables but wasn’t aware of the restriction!

I’ll change to 10897 (corresponds to “l” and “a”)

**Comment by svilupp** [comment]
This one is good to go.

I'll fix the SIGINT issue separately, but we first need to merge this.

**Review by marcom — APPROVED** [review]
